src/query/sql/src/planner/plans/scalar_expr.rs

```diff
@@ -663,6 +663,156 @@ pub trait Visitor<'a>: Sized {
     }
 }
 
+pub trait VisitorWithParent<'a>: Sized {
```

Comment:
Rename it to `VisitorWithRootScalarExpr` may improve the readability of the code